### PR TITLE
fix(app): Fix "run again" button after cancelling a run

### PR DESCRIPTION
**Pull request by mjhuff**
Closes [RQA-2982](https://opentrons.atlassian.net/browse/RQA-2982)

# Overview

After error recovery and drop tip changes, run context is required after the run is cancelled. Before, if the run was cancelled, the app would implicitly close the run context. We can't do that any longer, but in order for some setup buttons to work properly, we need to close the run context somehow. 

Currently, after a run is cancelled, the "run again" button is greyed out if a user hasn't calibrated their instruments, although they were just running the protocol prior to cancelling it (note that users can't actually play a run without calibration, but they can create a run). Let's allow users to "run again" (create a run), and then be in the proper run state to perform setup. I don't know if this ever was the intended behavior of protocol runs in the past, but a lot of the existing copy points to this behavior being desired, anyway. 

EDIT: After giving this more thought, while I do think we should make the "run again" button selectable to remain consistent with copy and behavioral expectations, it's possible to retain the old behavior as well and close the current run when run status is stopped. It requires more special casing within `ProtocolRunHeader`, but `isRunCurrent` is really something only utilized by drop tip CTAs at the end of the day, so it's not *too* difficult to reason about. Given the overall complexity of these interactions, I think it's best not to push our luck in assuming what does/doesn't work until we formally refactor all of this. 

TLDR: From an end user perspective, nothing changes between 7.X and 8.0. Functionally, we special-case the drop tip logic to play nicely with run currenting/uncurrenting, tie the new recovery error rendering logic to `isMostRecentRun` (effectively what we were doing in 7.X), and now give the client an escape hatch if the run is cancelled but doesn't close the current run context. 

## Test Plan and Hands on Testing
- Verified users can now calibrate instruments given the behavior described in the ticket. 

## Changelog
- The run again button is no longer disabled after a run is cancelled. 

## Risk assessment
low

[RQA-2982]: https://opentrons.atlassian.net/browse/RQA-2982?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/organisms/Devices/ProtocolRun/ProtocolRunHeader.tsx [hunk 1]
```diff
@@ -283,6 +285,11 @@ export function ProtocolRunHeader({
           ...robotAnalyticsData,
         },
       })
+
+      // Close the run if no tips are attached after running tip check at least once.
+      if (initialPipettesWithTipsCount === 0) {
+        closeCurrentRun()
+      }
```
Comment (SyntaxColoring) [inline app/src/organisms/Devices/ProtocolRun/ProtocolRunHeader.tsx]: Unless you think this will be understood by a frontend dev as-is, could you clarify in this comment why we're doing this?

**Review by SyntaxColoring — APPROVED** [review]
Discussed in-person. I have like a 30% understanding of the problem and I'm not really qualified to review the code, but it seems like this is tested and we have ideas to make it less confusing in the future, so LGTM.

File: app/src/organisms/Devices/ProtocolRun/ProtocolRunHeader.tsx (hunk 1 above)
Comment (mjhuff) [reply app/src/organisms/Devices/ProtocolRun/ProtocolRunHeader.tsx]: Yeah that's fair. Thanks!
